### ISSUE TITLE
Could add a min index value for `prefer-destructuring` for Array

### ISSUE THREAD

**Issue by mt-micky** [eslint/eslint#9386]
**Tell us about your environment**

* **ESLint Version:**
4.6.1
* **Node Version:**
6.11.2
* **npm Version:**
5.4.2

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**
air-bnb

**What did you do? Please include the actual source code causing the issue.**

```js
scope.vm.value = values[12];
```

**What did you expect to happen?**
Not complain about `prefer-destrucutring`

**What actually happened? Please include the actual, raw output from ESLint.**
I'm not sure what I'm suppose to write in this case.
The following seems silly to me:
```js
[,,,,,,,,,,,,scope.vm.value] = values;
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. For what it's worth, this case was discussed [here](https://github.com/eslint/eslint/pull/7741#discussion_r91868522) when implementing the rule, and we decided to just leave it as-is for now.

What is your use case for accessing the 12th element of an array? I'm having trouble understanding what situation would call for that.

**Comment by ljharb** [comment]
That is also my question; `[12]` seems like the kind of code smell that should require an eslint override, or a rearchitecting.

**Comment by mt-micky** [comment]
I picked `12` to emphasis all the commas (like the guy doing an example with 100 in the other issue)
The idea here was to allow an option to allow a max index.

Let's say `2` for example, so one could still write `const foo = args[2];`.
But won't be allowed to write `const bar = args[10]`, because, as @ljharb point it out, it sounds like bad coding somewhere anyways.

**Comment by not-an-aardvark** [comment]
I understand that, but I'm wondering about your use case because I'm currently not convinced that a max index is necessary. I think the level of silliness required to access such a high index with destructuring scales well with the fact that it generally shouldn't be necessary to access high indices anyway.

| Array Index | How often is it necessary to access this index, when writing good code? | How silly does it look to access this index with destructuring? |
|-------------|-------------------------------------------------------------------------|-----------------------------------------------------------------|
| 0           | Often                                                                   | Not silly                                                       |
| 1           | Somewhat rarely                                                         | Not silly                                                       |
| 2           | Very rarely                                                             | Mildly silly                                                    |
| ...         | ...                                                                     | ...                                                             |
| 13          | You should basically never do this                                      | Very silly                                                   |

**Comment by not-an-aardvark** [comment]
Can you fill out the [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/ee99876ccc76585b0af2c8673d7eb25bb0b972fa/templates/rule-change-proposal.md)? That might make some of the details of the proposal clearer.

**Comment by tjbenton** [comment]
I've come across this same thing when working with regex quite a bit.

```js
const [ , id ] = document.cookie.match(/JSESSIONID=([^;]+)/)
```
It seems silly to me to put an comma in there just to use destructuring.

The reason we even keep destructuring on there as preferred is to keep devs from doing something like this. Because destructuring handles this situation better.

```js
const foo = results[0]
const bar = results[1]
const baz = results[2]
```

The other place I've ran into this issue at is dealing with couchbase responses, and their keys, and dates.

**Comment by ljharb** [comment]
`const [, id] =` seems totally sensible to me. `match` returns a tupleN, and destructuring out the parts you want are exactly how one works with a tuple.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
